src/views/AddItem.jsx

```diff
@@ -1,3 +1,65 @@
+import { useState } from 'react';
+import { addItem } from '../api/firebase.js';
+
 export function AddItem() {
-	return <p>Hello from the <code>/add-item</code> page!</p>
+	const [itemName, setItemName] = useState('');
+	const [daysUntilNextPurchase, setDaysUntilNextPurchase] = useState(7);
+	const [itemAdded, setItemAdded] = useState(false);
+	const [error, setError] = useState(false);
+
+	const handleSubmit = async (e) => {
+		e.preventDefault();
+		try {
+			await addItem('my test list', {
+				itemName,
+				daysUntilNextPurchase,
+			});
+			setItemAdded(true);
+			setError(false);
+		} catch (e) {
+			setError(true);
+			setItemAdded(false);
+		}
+	};
+
+	return (
+		<>
+			<form onSubmit={handleSubmit}>
+				<label htmlFor="item-name-input">Item name:</label>
+				<input
+					type="text"
+					id="item-name-input"
+					onChange={(e) => setItemName(e.target.value)}
+				/>
+				<fieldset>
+					<legend>How soon will you buy this again?</legend>
+					<input
+						type="radio"
+						id="soon"
+						onChange={() => setDaysUntilNextPurchase(7)}
+						checked={daysUntilNextPurchase === 7}
+					/>
+					<label htmlFor="soon">Soon</label> <br />
+					<input
+						type="radio"
+						id="kind-of-soon"
+						onChange={() => setDaysUntilNextPurchase(14)}
+						checked={daysUntilNextPurchase === 14}
```

Comment:
Question out of curiosity, does like 47 work like insurance to ensure that the value of daysUntilNextPurchase is 14?

Reply:
Hi Beth. This is an awesome question, because it was hard for Emily and I to get what was going on as well. The `onChange` will update state when there is a change, but the `checked` property ensures that it is only checked when the value updated in state matches the value we assign. This makes it so that we can only check one radio button at a time (not two or three).  Here is more explanation on that: https://magnusbenoni.com/radio-buttons-react/

Reply:
And one EJ. 😂